freeipa/src/main/java/com/sequenceiq/freeipa/flow/freeipa/user/handler/SetPasswordHandler.java

```diff
@@ -43,7 +70,13 @@ public void accept(Event<SetPasswordRequest> setPasswordRequestEvent) {
             MDCBuilder.buildMdcContext(stack);
 
             FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
-            freeIpaClient.userSetPasswordWithExpiration(request.getUsername(), request.getPassword(), request.getExpirationInstant());
+            if (freeIpaClient.hasSetPasswordHashSupport()) {
+                LOGGER.info("IPA has password hash support, credentials information from UMS will be used.");
+                setPasswordHashFromUms(freeIpaClient, request.getUsername(), request.getUserCrn(), request.getExpirationInstant());
+            } else {
+                freeIpaClient.userSetPasswordWithExpiration(
```

Comment:
Let's add a log message here also. This will let us search to logs for whether anyone is using the "old way"